src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/DocumentState.cs

```diff
@@ -413,50 +414,61 @@ private async Task<(RazorCodeDocument, VersionStamp, VersionStamp)> GetGenerated
 
                 var codeDocument = projectEngine.ProcessDesignTime(documentSource, fileKind: document.FileKind, importSources, project.TagHelpers);
                 var csharpDocument = codeDocument.GetCSharpDocument();
+                var htmlDocument = codeDocument.GetHtmlDocument();
```

Comment:
Hmm, we may pay for this in older VS lol. It'll populate all the HTML documents for Blazor files in a project. I don't think we should do anything to change this now but we should definitely keep an eye on perf reports on how bad this hits performance via VS issues.

Reply:
Yeah. At this layer we won't know who is asking for the generated output. I thought about making the ToString() computation lazy but I didn't want to pre-optimize especially given we expect to have the __virtual.html in VS as well in the future.